README.md

```diff
@@ -24,7 +24,25 @@ A larger set of interesting patterns might include:
 * `^spew\.Dump$` -- forbid dumping detailed data to stdout
 * `^fmt\.Errorf(# please use github\.com/pkg/errors)?$` -- forbid Errorf, with a custom message
 
-Note that the linter has no knowledge of what packages were actually imported, so aliased imports will match these patterns.
+By default, patterns get matched against the actual expression as it appears in
+the source code. The effect is that ``^fmt\.Print.*$` will not match when that
+package gets imported with `import fmt2 "fmt"` and then the function gets
+called with `fmt2.Print`.
+
+This makes it hard to match packages that may get imported under a variety of
+different names, for example because there is no established convention or the
+name is so generic that import aliases have to be used. To solve this, a
+package that contains a subgroup literally called `pkg` (`(?P<pkg>...)`) will be
+matched against text where the import name got replaced with the full package
+name (e.g. `example.com/some/pkg`) if such a substitution is possible for the
+current expression. Otherwise such a pattern is ignored. This feature depends
+on additional information and only works in golangci-lint.
```

Comment:
can we support this locally?

Reply:
I suppose that depends on what your plans are for https://github.com/ashanbrown/forbidigo/blob/master/main.go.

Is it supposed to remain simple, with no dependency on "golang.org/x/tools/go/analysis"? If that dependency is acceptable, then a stand-alone analyzer with full support for all features is simple, similar to https://github.com/kubernetes-sigs/logtools/blob/main/logcheck/pkg/logcheck.go.

There is alreeady https://github.com/ashanbrown/forbidigo/blob/master/pkg/analyzer/analyzer.go, there's just no main.go for it. https://github.com/ashanbrown/forbidigo/pull/15 changes that.

Reply:
It looks to me like by adding `NeedsTypesInfo`  at https://github.com/ashanbrown/forbidigo/blob/master/main.go#L37, we'd be able to call `p.TypesInfo` at https://github.com/ashanbrown/forbidigo/blob/master/main.go#L58.

Reply:
(The reason I didn't want to switch to using the singlecheck analyzer is that it doesn't allow passing parameters as args on the command-line.  As it stands forbidigo is called as `forbidigo <patterns> [--] [packages...]`.   We'd be forced to pass them individually as `-p`  arguments if we merged that change.  But as mentioned above, my reading was that we don't need to use an analyzer to get TypesInfo data)

Reply:
This problem continues with https://github.com/ashanbrown/forbidigo/pull/22 needing `* types.Package`. I can check whether that can be obtained the same way.

Reply:
This worked out okay, see [other comment](https://github.com/ashanbrown/forbidigo/pull/21#discussion_r1023039808).